samples/getvmsbycluster.py

```diff
@@ -0,0 +1,182 @@
+    if summary['annotation'] is not '':
+        print("    Annotation: " + summary['annotation'])
+
+
+def vm2dict(dc, cluster, host, vm, summary):
+    # If nested folder path is required, split into a separate function
+    vmname = vm.summary.config.name
+    data[dc][cluster][host][vmname]['folder'] = vm.parent.name
+    data[dc][cluster][host][vmname]['mem'] = summary['mem']
+    data[dc][cluster][host][vmname]['disk'] = summary['disk']
+    data[dc][cluster][host][vmname]['cpu'] = summary['cpu']
+    data[dc][cluster][host][vmname]['path'] = summary['path']
+    data[dc][cluster][host][vmname]['ip'] = summary['ip']
+    data[dc][cluster][host][vmname]['guestname'] = summary['guestname']
+    data[dc][cluster][host][vmname]['state'] = summary['state']
+    data[dc][cluster][host][vmname]['managedby'] = summary['managedby']
+    data[dc][cluster][host][vmname]['annotation'] = summary['annotation']
+
+
+def data2csv(data, args):
+    with open(args.csvfile, 'w') as csvfile:
+        fieldnames = ['datacenter', 'cluster', 'host', 'folder', 'vmname',
+                      'disk', 'cpu', 'mem', 'path', 'guestname', 'state',
+                      'managedby', 'annotation', 'ip']
+        writer = csv.writer(csvfile)
+        writer.writerow(fieldnames)
+        for dc, clusters in data.iteritems():
+            for cl, hosts in clusters.iteritems():
+                for host, vms in hosts.iteritems():
+                    for vm in vms:
+                        currvm = data[dc][cl][host][vm]
+                        vmdata = [dc, cl, host, currvm['folder'], vm,
+                                  currvm['disk'], currvm['cpu'], currvm['mem'],
+                                  currvm['path'], currvm['guestname'],
+                                  currvm['state'], currvm['managedby'],
+                                  currvm['annotation'], currvm['ip']]
+                        writer.writerow(vmdata)
+
+
+def data2json(data, args):
+    with open(args.jsonfile, 'w') as f:
+        json.dump(data, f)
+
+
+def main():
+    """
+    Iterate through all datacenters and list VM info.
+    """
+    args = GetArgs()
+    json = True if args.json else False
+    csv = True if args.csv else False
+
+    if args.password:
+        password = args.password
+    else:
+        password = getpass.getpass(prompt='Enter password for host %s and '
+                                   'user %s: ' % (args.host, args.user))
+
+    si = SmartConnect(host=args.host,
+                      user=args.user,
+                      pwd=password,
+                      port=int(args.port))
+    if not si:
+        print("Could not connect to the specified host using specified "
+              "username and password")
+        return -1
+
+    atexit.register(Disconnect, si)
+
+    content = si.RetrieveContent()
+    children = content.rootFolder.childEntity
+    for child in children:  # Iterate though DataCenters
+        if hasattr(child, 'hostFolder'):
```

Comment:
I believe there is only one hostFolder under datacenter, would it be better to avoid the iteration and just open hostFolder?